pyproject.toml

```
@@ -109,8 +109,7 @@ explicit = true
 
 [tool.uv.sources]
 # ghex = { git = "https://github.com/ghex-org/GHEX.git", branch = "master"}
-# TODO(egparedes): use GT4Py tagged release until #596 & gt4py#1738 are merged
-gt4py = { git = "https://github.com/GridTools/gt4py", tag = "icon4py_20241113" }
+gt4py = { git = "https://github.com/GridTools/gt4py.git", branch = "main" }
```

I took the freedom to update this PR and merge the latest icon4py main, in order to include the `uv` development environment.

While merging, I reverted a change from @havogt, that used a custom gt4py branch. If needed, you can re-apply it by changing this line:
```suggestion
gt4py = { git = "https://github.com/havogt/gt4py.git", branch = "increase_inlining_loop" }
```